### PR TITLE
Feature flag sync

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206046777189412/f 

### Description
Add Remote flags around sync.

### Steps to test this PR
https://app.asana.com/0/0/1206190385973972/f

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

**Comment by cmonfortep** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #4019** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4019?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4019?utm_source=stack-comment).

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncDisabledViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+@SuppressLint("NoLifecycleObserver") // does not subscribe to app lifecycle
+class SyncDisabledViewModel(
+    private val syncFeatureToggle: SyncFeatureToggle,
+    private val syncStateMonitor: SyncStateMonitor,
+    private val dispatcherProvider: DispatcherProvider,
+) : ViewModel(), DefaultLifecycleObserver {
+
+    data class ViewState(
+        val warningMessage: Int? = null,
```
Comment (malmstein) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncDisabledViewModel.kt:45]: nit 
```suggestion
        val message: Int? = null,
```

File: sync/sync-impl/src/main/res/layout/view_sync_enabled.xml [hunk 2]
```diff
@@ -103,7 +107,8 @@
         app:leadingIcon="@drawable/ic_camera_photos_24"
         app:leadingIconBackground="circular"
         app:primaryText="@string/sync_show_qr_code_item"
-        app:primaryTextTruncated="true" />
+        app:primaryTextTruncated="true"
+        app:secondaryText=""/>
```
Comment (malmstein) [inline sync/sync-impl/src/main/res/layout/view_sync_enabled.xml:111]: OneLineListItem doesn’t have a secondary text. you can remove this

**Review by malmstein — APPROVED** [review]
besides what we’ve seen in the review task, this is great. Good job @cmonfortep !

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/engine/RealSyncEngine.kt [hunk 3]
```diff
@@ -61,7 +61,7 @@ class RealSyncEngine @Inject constructor(
 
     override fun triggerSync(trigger: SyncTrigger) {
         Timber.i("Sync-Engine: petition to sync now trigger: $trigger")
-        if (syncStore.isSignedIn()) {
+        if (syncStore.isSignedIn() && syncStore.syncingDataEnabled) {
```
[the remark this answers is not in the document]
Comment (cmonfortep) [reply sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/engine/RealSyncEngine.kt:64]: We could do that, but we still need to keep `isSignedIn` and `syncingDataEnabled` for other usages, so we are not having any extra benefit. Will leave it like this for now.

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncFeature.kt [hunk 4]
```diff
@@ -22,9 +22,22 @@ import com.duckduckgo.feature.toggles.api.Toggle
...
-    featureName = "deviceSync",
+    featureName = "sync",
 )
 interface SyncFeature {
     @Toggle.DefaultValue(false)
+    @Toggle.InternalAlwaysEnabled
     fun self(): Toggle
+
+    @Toggle.DefaultValue(true)
+    fun level0ShowSync(): Toggle
```
[the remark this answers is not in the document]
Comment (cmonfortep) [reply sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncFeature.kt:33]: needs to follow same name as in privacy config, so we can't change that.

**Pull request merged**